torchvision/models/detection/faster_rcnn.py

```diff
@@ -224,11 +226,8 @@ def __init__(self, backbone, num_classes=None,
             bbox_reg_weights,
             box_score_thresh, box_nms_thresh, box_detections_per_img)
 
-        if image_mean is None:
-            image_mean = [0.485, 0.456, 0.406]
-        if image_std is None:
-            image_std = [0.229, 0.224, 0.225]
-        transform = GeneralizedRCNNTransform(min_size, max_size, image_mean, image_std)
+        if transform is None:
+            transform = GeneralizedRCNNTransform(min_size, max_size, image_mean, image_std)
```

Comment:
My comment before corresponds to the case where `transform is not None` but `min_size is not None` (which we should change the default to be `None`)
We should raise an error in this case.

Something in the lines of
```python
if transform is None:
    if min_size is None:
        min_size = 800
    ...
else:
    if min_size is not None:
        raise ValueError("Can't specify both transform and min_size")
```
The same for `max_size`, `image_mean` and `image_std`.

The reason why it would be good to make this check is that the user could have a very different type of custom transform, for which `max_size` is non-existing. In this case, specifying a `max_size` would just be confusing, so we should either use `transform` or the default parameters.